spec/lib/whitehall_importer/change_history_spec.rb

```diff
@@ -0,0 +1,81 @@
+RSpec.describe WhitehallImporter::ChangeHistory do
+  let(:first_edition) do
+    build(:whitehall_export_edition,
+          state: "superseded",
+          revision_history: [build(:whitehall_export_revision_history_event,
+                                   state: "published",
+                                   created_at: Time.zone.now.last_year.rfc3339)])
+  end
+
+  it "returns change history for the published edition" do
+    superseded_edition = build(:whitehall_export_edition,
+                               state: "superseded",
+                               change_note: "Some fantastic changes",
+                               revision_history: [build(:whitehall_export_revision_history_event,
+                                                        state: "published",
+                                                        created_at: Time.zone.now.beginning_of_day.rfc3339)])
+    published_edition = build(:whitehall_export_edition,
+                              state: "published",
+                              change_note: "Some great changes",
+                              revision_history: [build(:whitehall_export_revision_history_event,
+                                                       state: "published")])
+    whitehall_export = build(:whitehall_export_document,
+                             editions: [first_edition, superseded_edition, published_edition])
+
+    expect(described_class.new(whitehall_export).for(3)).to match([
+      { "id" => anything, "note" => "Some fantastic changes", "public_timestamp" => Time.zone.now.beginning_of_day.rfc3339 },
+    ])
+  end
+
+  it "returns draft edition change history if it has been previously unpublished" do
+    edition = build(:whitehall_export_edition,
+                    state: "draft",
+                    change_note: "Some great changes",
+                    unpublishing: build(:whitehall_export_unpublishing),
+                    revision_history: [build(:whitehall_export_revision_history_event,
+                                             state: "published",
+                                             created_at: Time.zone.now.beginning_of_day.rfc3339)])
+    whitehall_export = build(:whitehall_export_document,
+                             editions: [first_edition, edition])
+
+    expect(described_class.new(whitehall_export).for(3)).to match([
+      { "id" => anything, "note" => "Some great changes", "public_timestamp" => Time.zone.now.beginning_of_day.rfc3339 },
+    ])
+  end
+
+  it "doesn't return change history for minor change editions" do
```

Comment:
You probably also want a test that it skips drafts.

Reply:
Sure!